spec/models/communication_spec.rb

```diff
@@ -15,21 +15,54 @@
     it_behaves_like "a record that is deletable"
   end
 
-  describe ".create_with_twilio_details!" do
-    let(:appointment) { create(:appointment) }
-
-    it "creates a communication with a TwilioSmsDeliveryDetail" do
-      expect {
-        Communication.create_with_twilio_details!(appointment: appointment,
-                                                  twilio_sid: SecureRandom.uuid,
-                                                  twilio_msg_status: "sent",
-                                                  communication_type: :missed_visit_sms_reminder)
-      }.to change { Communication.count }.by(1)
-        .and change { TwilioSmsDeliveryDetail.count }.by(1)
+  context "Utilities" do
+    describe ".next_appointment_time" do
+      before do
+        @original_time_zone = Time.zone
+        Time.zone = Rails.application.config.country[:time_zone]
+        allow(Communication).to receive(:messaging_start_hour).and_return(10)
+        allow(Communication).to receive(:messaging_end_hour).and_return(16)
+      end
+
+      after do
```

Comment:
This could be an `around` do block with a `Time.use_zone` instead. That'll be quite neat.